components/dropdown/dropdown-menu.js

```diff
@@ -68,11 +77,19 @@ class DropdownMenu extends ThemeMixin(DropdownContentMixin(LitElement)) {
 		menu.show({ preventFocus: true });
 	}
 
-	_onFocus() {
+	_onFocus(e) {
+		if (e.srcElement === this) return;
```

Comment:
I've now got a hypothesis for why adding the close button causes that focus loop! The close button is NOT in the footer slot, so its focus event bubbles up the DOM to the dropdown menu when the button gets focused on. This is picked up by the onFocus method of dropdown-menu , which proceeds to override the focus to the menu, causing that loop. We need to ignore focus events coming from the dropdown element. I tested the existing dropdown menus with this change, and this return case is only hit with the close button.

Reply:
Nice, that makes sense!